Build fails with older CMake versions

The OpenSSL external project fails with older CMake versions with a cryptic error. On Linux it looks like this (with CMake 3.5.1):
```
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[  1%] No patch step for 'openssl'
make[2]: *** No rule to make target 'openssl/src/openssl-stamp/openssl-skip-update', needed by 'openssl/src/openssl-stamp/openssl-configure'. Stop.
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/openssl.dir/all' failed
make[1]: *** [CMakeFiles/openssl.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

Not exactly sure what causes the failure, but we should bump our `cmake_minimum_required` accordingly.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,5 @@
 language: c
 
-addons:
-  apt:
-    sources:
-    - kalakris-cmake
-    packages:
-    - cmake
-
 sudo: false
 
 env:

--- a/deps/openssl.cmake
+++ b/deps/openssl.cmake
@@ -38,8 +38,6 @@ else (WithSharedOpenSSL)
       CONFIGURE_COMMAND ${OPENSSL_CONFIGURE_COMMAND}
       INSTALL_COMMAND   ""
       TEST_COMMAND      ""
-      UPDATE_COMMAND    ""
-      UPDATE_DISCONNECTED YES
   )
   
   set(OPENSSL_DIR ${CMAKE_BINARY_DIR}/openssl/src/openssl)

